homeassistant/components/deconz/binary_sensor.py

```diff
@@ -104,11 +143,18 @@ def async_add_sensor(
             ):
                 entities.append(DeconzBinarySensor(sensor, gateway))
 
-            if sensor.tampered is not None:
-                known_tampering_sensors = set(gateway.entities[DOMAIN])
-                new_tampering_sensor = DeconzTampering(sensor, gateway)
-                if new_tampering_sensor.unique_id not in known_tampering_sensors:
-                    entities.append(new_tampering_sensor)
+            for sensor_description in BINARY_SENSOR_DESCRIPTIONS:
+
+                try:
+                    if sensor_description.value_fn(sensor) is not None:
+                        known_sensors = set(gateway.entities[DOMAIN])
+                        new_sensor = DeconzPropertyBinarySensor(
+                            sensor, gateway, sensor_description
+                        )
+                        if new_sensor.unique_id not in known_sensors:
+                            entities.append(new_sensor)
+                except AttributeError:
+                    continue
```

Comment:
Normally we move code out of the `try` block that isn't expected to generate `AttributeError`

Reply:
Better like this?

Reply:
Yes. Personal preference would be to refactor to avoid a try/except here, but thats not a blocking issue

Reply:
Do you have a suggestion? I removed using has_attr to verify the existence of the attribute, should I create a second method to check its existence?

Reply:
Maybe something like 

```
--- a/homeassistant/components/deconz/binary_sensor.py
+++ b/homeassistant/components/deconz/binary_sensor.py
@@ -56,6 +56,7 @@ class DeconzBinarySensorDescriptionMixin:
 
     suffix: str
     update_key: str
+    required_attr: str
     value_fn: Callable[[PydeconzSensor], bool | None]
 
 
@@ -98,6 +99,7 @@ ENTITY_DESCRIPTIONS = {
 BINARY_SENSOR_DESCRIPTIONS = [
     DeconzBinarySensorDescription(
         key="tamper",
+        required_attr="tampered",
         value_fn=lambda device: device.tampered,
         suffix="Tampered",
         update_key="tampered",
@@ -106,6 +108,7 @@ BINARY_SENSOR_DESCRIPTIONS = [
     ),
     DeconzBinarySensorDescription(
         key="low_battery",
+        required_attr="low_battery",
         value_fn=lambda device: device.low_battery,
         suffix="Low Battery",
         update_key="lowbattery",
@@ -144,11 +147,7 @@ async def async_setup_entry(
                 entities.append(DeconzBinarySensor(sensor, gateway))
 
             for sensor_description in BINARY_SENSOR_DESCRIPTIONS:
-
-                try:
-                    if sensor_description.value_fn(sensor) is None:
-                        continue
-                except AttributeError:
+                if not hasattr(sensor, sensor_description.required_attr):
                     continue
 
                 known_sensors = set(gateway.entities[DOMAIN])
```

Reply:
Done, I also did the same changes to the sensor platform that got merged earlier tonight